src/refresh-runtime.js

```diff
@@ -581,41 +581,60 @@ function debounce(fn, delay) {
   };
 }
 
-const enqueueUpdate = debounce(performReactRefresh, 16);
+const enqueueUpdate = debounce(() => {
+  window.__beforePerformReactRefresh?.();
```

Comment:
I guess it'd nice to support registering multiple functions.
For example:
```js
const hooks = []
window.__registerBeforePerformReactRefresh = (cb) => {
  hooks.push(cb)
}

// run in enqueueUpdate
hooks.forEach(hook => hook())

// framework code

window.__registerBeforePerformReactRefresh(() => {
  // do something that's needed
})
```

Another way to expose this functionality is to use an exported function:
```js
import { registerBeforePerformReactRefreshHook } from '/@react-refresh-api'

registerBeforePerformReactRefreshHook(() => {
  // do something that's needed for that framework
})
```
I'm not sure which is better.